lib/libc/minimal/include/ctype.h

```diff
@@ -15,54 +15,58 @@ extern "C" {
 
 static inline int isupper(int a)
 {
-	return ((unsigned)(a)-'A') < 26;
+	return (int)(((unsigned)(a)-(unsigned)'A') < 26U);
 }
```

Comment:
I don't see why that outer cast was added.  The result of a "<" operator is int by definition per the standard, no?

Reply:
the problem is that the result of "<" is a boolean (According with MISRA, C standard is clear saying that this return an int) and the function return an int